src/BenchmarkDotNet/Validators/DotNetSdkVersionValidator.cs

```diff
@@ -0,0 +1,326 @@
+                            {
+                                versions.Add(name);
+                            }
+                            else
+                            {
+                                if (!(string.IsNullOrEmpty(sp)) && install == "1")
+                                {
+                                    versions.Add(name + " SP" + sp);
+                                }
+                                else if (install == "1")
+                                {
+                                    versions.Add(name);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "This code is protected with a runtime OS platform check")]
+        private static void Get45PlusFromRegistry(List<string> versions)
+        {
+            const string subkey = @"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\";
+
+            using (var ndpKey = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(subkey))
+            {
+                if (ndpKey == null)
+                {
+                    return;
+                }
+
+                if (ndpKey.GetValue("Version") != null)
+                {
+                    versions.Add(ndpKey.GetValue("Version").ToString());
+                }
+                else
+                {
+                    if (ndpKey.GetValue("Release") != null)
+                    {
+                        versions.Add(CheckFor45PlusVersion((int)ndpKey.GetValue("Release")));
+                    }
+                }
+            }
+        }
+
+        private static string CheckFor45PlusVersion(int releaseKey)
+        {
+            if (releaseKey >= 533320)
+                return "4.8.1";
+            if (releaseKey >= 528040)
+                return "4.8";
+            if (releaseKey >= 461808)
+                return "4.7.2";
+            if (releaseKey >= 461308)
+                return "4.7.1";
+            if (releaseKey >= 460798)
+                return "4.7";
+            if (releaseKey >= 394802)
+                return "4.6.2";
+            if (releaseKey >= 394254)
+                return "4.6.1";
+            if (releaseKey >= 393295)
+                return "4.6";
+            if (releaseKey >= 379893)
+                return "4.5.2";
+            if (releaseKey >= 378675)
+                return "4.5.1";
+            if (releaseKey >= 378389)
+                return "4.5";
+
+            return "";
+        }
+
+        private static string GetSdkVersionFromMoniker(RuntimeMoniker runtimeMoniker)
+        {
+            switch (runtimeMoniker)
+            {
+                case RuntimeMoniker.Net461:
+                    return "4.6.1";
+
+                case RuntimeMoniker.Net462:
+                    return "4.6.2";
+
```

Comment:
Nit: this could be more compact without reducing legibility.
```cs
            return runtimeMoniker switch
            {
                RuntimeMoniker.Net461 => "4.6.1",
                RuntimeMoniker.Net462 => "4.6.2",

```